x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizerTests.java

```diff
@@ -206,16 +212,25 @@ public void testSingleFieldExtractor() {
         var filter = as(limit.child(), FilterExec.class);
         var extract = as(filter.child(), FieldExtractExec.class);
 
-        assertEquals(
-            Sets.difference(allFieldsExcludingTextWithExact(mapping), Set.of("emp_no")),
-            Sets.newHashSet(names(restExtract.attributesToExtract()))
-        );
+        assertEquals(Sets.difference(allFields(mapping), Set.of("emp_no")), Sets.newHashSet(names(restExtract.attributesToExtract())));
         assertEquals(Set.of("emp_no"), Sets.newHashSet(names(extract.attributesToExtract())));
 
         var query = as(extract.child(), EsQueryExec.class);
         assertThat(query.estimatedRowSize(), equalTo(Integer.BYTES + allFieldRowSize));
     }
 
+    private Set<String> allFields(Map<String, EsField> mapping) {
```

Comment:
What purpose does this method serve; in particular where it is being used?

Reply:
It's used in two tests that check the output based on the mapping. 
Before this change, it was just `mapping.keySet()`, but now the mapping also contains a TEXT field with a KEYWORD subfield, so we have to take it into consideration as well.